templates/openstackclient/bin/tripleo-deploy.sh

```diff
@@ -120,11 +121,48 @@ play() {
     sed -i -e 's/OSP_DIRECTOR_OPERATOR_DEPLOY_IDENTIFIER/'${DEPLOY_IDENTIFIER}'/g' $i
   done
 
+  STACK_ACTION_ARG=""
+  if [ "${OSP_VERSION}" = "16.2" ]; then
+    # We need stack_action to be set correctly in 16.2 as the network_config and octavia roles in tripleo-ansible
+    # rely on it.
+    # As we can't use the existance of a heat stack, instead use the existance of /var/lib/tripleo-config on any
+    # overcloud host (created by the tripleo-bootstrap role) to determine if this is a "CREATE" or an "UPDATE"
+    cat <<EOF > set_stack_action_playbook.yaml
+{{` ---
+- hosts: overcloud
+  name: Determine the correct stack action for 16.2 deployments
+  gather_facts: false
+  tasks:
+    - stat:
```

Comment:
So, we check any node on the overcloud for the existence of this directory, if it exists we set the stack_action: UPDATE. 

I wonder if this would cause issues for nodes being added to the stack? In that case, Heat's StackAction would be set to CREATE. So I wonder if forcing it to UPDATE globally would cause issues for those nodes? Off the top of my head, I think it should be fine, since we primarily use it to do specific things on the Controller nodes, the Computes should be ok.

Looking at the network-config component, it should be fine if the stack_action is UPDATE since we will also check to see if it has been run before, and if not we will run os-net-config:

https://github.com/openstack/tripleo-ansible/blob/stable/train/tripleo_ansible/roles/tripleo-network-config/tasks/main.yml#L113-L128